rcore-fs-sefs/src/lib.rs

```diff
@@ -207,6 +210,131 @@ impl INodeImpl {
         self.fs.meta_file.flush()?;
         Ok(())
     }
+
+    fn zero_range(
+        &self,
+        start_offset: usize,
+        end_offset: usize,
+        keep_size: bool,
+    ) -> vfs::Result<()> {
+        assert!(start_offset < end_offset);
+        let mut inode = self.disk_inode.write();
+        let file_size = inode.size as usize;
+        let (mut offset, mut len) = {
+            let (start, end) = if keep_size {
+                (start_offset.min(file_size), end_offset.min(file_size))
+            } else {
+                (start_offset, end_offset)
+            };
+            (start, end - start)
+        };
+        static ZEROS: [u8; BLKSIZE] = [0; BLKSIZE];
+        while len != 0 {
+            let len_per_loop = len.min(BLKSIZE);
+            let len_written = self
+                .file
+                .write_at(&ZEROS[..len_per_loop], offset)
+                .expect("failed to write zeros");
+            len -= len_written;
+            offset += len_written;
+        }
```

Comment:
I guess these ten lines of code have appeared repeatedly; not only this file, but other files in sefs as well. One trick to avoid repeating yourself is to implement extension traits. In this specific use case, you can implement an extension trait that have a `write_zeros` utility method.

I also found it very convenient to add utility methods like `write_exact` and `read_exact` to files. See [the `SocketStreamExt` trait](https://github.com/occlum/ngo/blob/master/src/libos/crates/host-socket/tests/echo.rs#L328) in NGO.

Reply:
Good suggestion, add `write_zeros_at` for the backed file of sefs